shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -88,56 +89,54 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
             Session session = setupSegments.get(new SimpleImmutableEntry<>(append.segment, append.getWriterId()));
             validateAppend(append, session);
             if (!append.segment.equals(segmentBeingAppendedTo) || !append.getWriterId().equals(writerIdPerformingAppends)) {
-                breakFromAppend(out);
+                breakFromAppend(null, null, out);
+            }
+            ByteBuf data = append.getData().slice();
```

Comment:
But, this will affect the user supplied buffer. 
I saw this one ByteArraySerializer .. Byte Array is used as it is for ByteBuf.wrap which does not create any copy. so, its quite dangerous use supplied buffer as it is.

Reply:
however, if we take care the ByteArraySerializer  to take the byte copy , then we can remove the .slice() from the CommandEncoder. 
But, I will do it as separate PR ;  changing this requires changes in the unit tests also

Reply:
`ByteBuf.slice()` creates a derived buffer which shares the memory region of the parent buffer.

Reply:
but ,its copy.  if we dont use the .slice() then whatever changes to data will reflect the append.Data.

Reply:
ok .. got it : https://netty.io/4.1/api/io/netty/buffer/ByteBuf.html#slice--

no duplicate buffer ; but indexes are separate; then we can keep the slice in CommandEncoder its good.. no additional buffer only indices differs